test/integration/scheduler/scheduler_test.go

```diff
@@ -611,3 +619,174 @@ func TestNodeEvents(t *testing.T) {
 	}
 
 }
+
+// TestPodSchedulingContextSSA checks that the dynamicresources plugin falls
+// back to SSA successfully when the normal Update call encountered
+// a conflict.
+//
+// This is an integration test because:
+//   - Unit testing does not cover RBAC rules.
+//   - Triggering this particular race is harder in E2E testing
+//     and harder to verify (needs apiserver metrics and there's
+//     no standard API for those).
+func TestPodSchedulingContextSSA(t *testing.T) {
```

Comment:
Triggering the race here isn't trivial either. I'd appreciate feedback on the current approach.  In parallel I'll investigate whether error injection can be used to trigger the fallback.

Reply:
I've pushed a second commit that does error injection at the `http.RoundTripper` level.

@aojea: you've done some work on integration testing and apiserver setup. Do my changes look okay to you?

/hold

I want to squash before merging.

Reply:
It can, and I prefer the new approach. I just haven't squashed yet to simplify before/after comparisons (if anyone cares).

Reply:
Scratch that - I also fixes some other issues, squashed and force-pushed as a single commit.

Reply:
checking now sorry, @pohly  you can so something like

```
diff --git a/test/integration/framework/test_server.go b/test/integration/framework/test_server.go
index 6b15b09b2d8..d288ff73ca8 100644
--- a/test/integration/framework/test_server.go
+++ b/test/integration/framework/test_server.go
@@ -52,8 +52,9 @@ AwEHoUQDQgAEH6cuzP8XuD5wal6wf9M6xDljTOPLX2i8uIp/C/ASqiIGUeeKQtX0
 
 // TestServerSetup holds configuration information for a kube-apiserver test server.
 type TestServerSetup struct {
-       ModifyServerRunOptions func(*options.ServerRunOptions)
-       ModifyServerConfig     func(*controlplane.Config)
+       ModifyServerRunOptions   func(*options.ServerRunOptions)
+       ModifyServerConfig       func(*controlplane.Config)
+       ModifyServerClientConfig func(*rest.Config)
 }
 
 type TearDownFunc func()
@@ -223,6 +224,7 @@ func StartTestServer(ctx context.Context, t testing.TB, setup TestServerSetup) (
                t.Fatal(err)
        }
 
+       setup.ModifyServerClientConfig(kubeAPIServerClientConfig)
        kubeAPIServerClient, err := client.NewForConfig(kubeAPIServerClientConfig)
        if err != nil {
                t.Fatal(err)
````

and then use it like
```
	_, kubeConfig, tearDownFn := framework.StartTestServer(ctx, t, framework.TestServerSetup{
		ModifyServerRunOptions: func(opts *options.ServerRunOptions) {
			// Disable ServiceAccount admission plugin as we don't have serviceaccount controller running.
			opts.Admission.GenericAdmission.DisablePlugins = []string{"ServiceAccount", "TaintNodesByCondition"}
			opts.APIEnablement.RuntimeConfig.Set("networking.k8s.io/v1alpha1=true")
		},
		ModifyServerClientConfig: func(c *clientrest.Config) {
			c.Wrap(func(rt http.RoundTripper) http.RoundTripper {
				return roundTripperFunc(func(req *http.Request) (*http.Response, error) {
					authorizationHeaderValues.append(req.Header.Values("Authorization"))
					return rt.RoundTrip(req)
				})
			})
		},
	})
	defer tearDownFn()
```

Reply:
Thanks for the suggestions. I removed all changes from `test/integration/scheduler/scheduler_test.go`.